pkg/trait/camel.go

```diff
@@ -276,5 +278,5 @@ func determineRuntimeVersion(e *Environment) string {
 	if e.Platform != nil && e.Platform.Status.Build.RuntimeVersion != "" {
 		return e.Platform.Status.Build.RuntimeVersion
 	}
-	return ""
```

Comment:
I think this was on purpose. When there is no runtime set in any of the status, it means something wrong happened. I think we need to keep the empty value here to avoid unpredictable errors later. You can see the `Configure()` func checking it and returning an error when this situation arises. Probably a comment on top of the func would be good to avoid future misunderstanding.

Reply:
`determineRuntimeVersion()` now returns an error for this case. I think it is more clear this way